### DISCUSSION TITLE
info hook avoiding extend session

### DISCUSSION THREAD

**Discussion by fnieri-cdp** [OpenIDC/mod_auth_openidc#1279] [Q&A]
I read that from release 2.4.14.2 there was the support for `extend_session=false` query parameter to the info hook to avoid extending the session on calls to the info hook.
I'm using release  2.4.14.4 and I assumed calling `auth/redirect_uri?info=json&extend_session=false`, the `timeout` value in the json response wasn't going to change. Instead repetitively calling the previous endpoint, every minutes the value of attribute `timeout `changes showing a session renewal.
Can anyone tell me if I'm doing something wrong, I assumed incorrect behavior or if the feature is no longer present?

Thanks

**Comment by zandbelt** [comment]
this is a pretty much undocumented feature with a clumsy implementation so far: the functionality works because the session is not written to the cache... ; however, the inactivity timeout value in the current session that is loaded in memory is updated and represented as such in the response from the info hook; but since it is never saved to cache, the actual timeout will occur as expected on subsequent requests (that load the session from cache with the original value)

**Reply by fnieri-cdp** [reply]
Thanks for the answer!

So is there a way to periodically retrieve from the web browser via ajax the information of when the session will expire?
For example to warn the user about expiration can be usefull expecially during form compilation than can't preserved in POST request (ex. attached files).
It would have been optimal for the response json of the info hook to contain the real session timeout value.

**Reply by zandbelt** [reply]
I agree, we'll look into it

**Comment by zandbelt** [comment]
see fed0f2f71bde4143cb0d48bc792265f5dfdec0de

**Reply by fnieri-cdp** [reply]
Thank you very much, I didn't expect so quickly

**Comment by zandbelt** [comment]
https://github.com/OpenIDC/mod_auth_openidc/releases/tag/v2.4.16.6
